mern-ecommerce/controllers/user.js

```diff
@@ -40,6 +41,20 @@ exports.update = (req, res) => {
   );
 };
 
+exports.productById = (req, res, next, id) => { 
+  Product.findById(id) 
+    .populate('products.product', 'name price')
+    .exec((err, product) => { 
```

Comment:
The callback function in the `exec` method should have a consistent naming convention for error and result variables. Instead of `err` and `product`, consider using `error` and `result` or `err` and `data` throughout the code.